### ISSUE TITLE
Documentation for default values is not accurate

### ISSUE THREAD

**Issue by AlekSi** [golangci/golangci-lint#2170]
### Your feature request related to a problem? Please describe.

I was reading https://golangci-lint.run/usage/configuration/ when I noticed a problem:
```yaml
# This file contains all available configuration options
# with their default values.

# ...

nestif:
    # minimal complexity of if statements to report, 5 by default
    min-complexity: 4
```

The comment on top of the file ("with their default values") contradicts the comment for `nestif.min-complexity` field.

### Describe the solution you'd like.

1. Remove default values from comments.
2. Put default values into field values.
3. Write a test that unmarshals `.golangci.example.yml` and compares results with `defaultLintersSettings`, failing if they are different.

### Describe alternatives you've considered.

One alternative would be to fix everything manually.
Another would be to generate `.golangci.example.yml`, but that's likely to be tricky due to comments.
Lastly, we could just remove the comment on top of the file, but that would make the documentation less useful.

**Comment by AlekSi** [comment]
> This file contains all available configuration options with their default values.

Ok, then what does it mean?

**Comment by ldez** [comment]
> This file contains all available configuration options

This file provides an exhaustive list of the options of the linters

> with their default values.

in comments

**Comment by AlekSi** [comment]
What about options without default values in comments? For example, what are the default values for errorlint or godot linters? How can a typical user learn them?

I don't think the current documentation is very helpful in that regard, and I propose to make it better without less manual work and more automation.
